### DISCUSSION TITLE
Merge or rebase?

### DISCUSSION THREAD

**Discussion by lassik** [scheme-live/live#35] [General]
Should we use merge or rebase?

Personally, I prefer rebase since I find the linear history easier to read.

**Comment by amirouche** [comment]
:shrug:

**Comment by amirouche** [comment]
The problem with rebase is that it is more immediate work, whereas merge is more often less painful. Also who reads history as graph?

**Comment by christaus** [comment]
Rebase or merge are more related to the developers who use them. This is also related to the project.
You can rebase something to keep a clear roadmap.
Merging don't mean you have something unclean, you can merge if you want to keep a track of different changes on branch.
To me merging or rebasing is to use case by case not as a global solution, both can be usefull.
A good decision tool could be gitk or git graph for vscodium/code . This way you have a global visual handle of your entire project branchs.
